### PR TITLE
TokenMigrator: Prevent rate == 0

### LINKED ISSUE #983: Prevent rate == 0 in TokenMigrator.sol

### PR DESCRIPTION
**Pull request by nicholaspai**
- Prevent rate being 0 to prevent `migrateTokens` from dividing by 0
- Change migrated new token amount to use `divCeil` instead of `div`, so that repeating decimals don't cause token holders to have fewer tokens than necessary, and most importantly doesn't cause really small token holders (e.g. 1 wei) to get floored down to 0 tokens.

Fixes #983 

Signed-off-by: Nick Pai <npai.nyc@gmail.com>

### CODE REVIEW

File: core/contracts/oracle/implementation/TokenMigrator.sol [hunk 1]
```diff
@@ -48,7 +51,7 @@ contract TokenMigrator {
             return;
         }
 
-        FixedPoint.Unsigned memory newBalance = oldBalance.div(rate);
+        FixedPoint.Unsigned memory newBalance = oldBalance.divCeil(rate);
```
Comment (mrice32) [inline core/contracts/oracle/implementation/TokenMigrator.sol]: Not necessarily against this change, but curious why you chose to ceil this?

One minor issue I see with ceiling this is that if rate is >> 1, rounding errors could allow token holders to split up their tokens (pre-snapshot) in such a way to generate "free" new tokens. In most cases 1 wei of new tokens wouldn't be worth the gas for an extra transaction, but theoretically there are cases where that amount might matter. On the other hand, if we round down, you can only lose tokens by splitting them up.

Curious to hear everyone's thoughts. @ptare @chrismaree

Comment (nicholaspai) [reply core/contracts/oracle/implementation/TokenMigrator.sol]: @mrice32 interesting I hadn't thought of that. To be honest, this change was only to mitigate downside for tiny token holders. In practice, it wouldn't really be an issue and would more often result in normal token holders gaining an extra wei of tokens via rounding. Happy to revert this but curious what @ptare and @chrismaree think

Comment (nicholaspai) [reply core/contracts/oracle/implementation/TokenMigrator.sol]: Thinking more about this, I think this `divCeil` creates more upside risk (token holders getting more tokens than they should post-migration) than downside risk, I'm going to revert back to `div` but still curious what @ptare and @chrismaree opine

Comment (ptare) [reply core/contracts/oracle/implementation/TokenMigrator.sol]: I don't have a strong opinion, probably won't matter much either way.

Comment (chrismaree) [reply core/contracts/oracle/implementation/TokenMigrator.sol]: ye, I agree that `div` makes the most sense here but also not super strong feelings.

**Pull request merged**